### ISSUE TITLE
AbstractControl has/getError for multiple classes of errors

### ISSUE THREAD

**Issue by Barbiero** [angular/angular#25553]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ X ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

Currently the `getError` and `hasError` functions on [AbstractControl](https://angular.io/api/forms/AbstractControl) have the following signature:
```typescript
hasError(errorCode: string, path?: string[]): boolean
```
However, it is not uncommon for certain classes of error to result in the same message, especially so with the `minlegth` behavior for allowing optional selections:
```html
<mat-error *ngIf="form.hasError('minlength', ['documents']) || form.hasError('required', ['documents'])">
  Must select at least 4 documents
</mat-error>
```
## Proposal
So my proposal would be to change `hasError` to the following signature:
```typescript
hasError(errorCode: string | string[], path?: string[]): boolean
```
and `getError` to
```typescript
getError(errorCode: string | string[], path?: string[]): any
```
in the sense that I would be able to call `form.hasError(['minlength','required'], ['documents'])` so that I could confirm that at least one of those errors are applied.

Another approach would be to create new functions, namely `hasSomeErrors` and  `hasAllErrors`(and their `get` equivalents, maybe?) that would implement this behavior for logical disjunction("OR") and logical conjunction("AND") for the list of `errorCode`s passed to its parameters. This approach would not change behavior for existing functions. I'd like some discussion here since I'm not sure which approach would be the best.

## What is the motivation / use case for changing the behavior?

Simplifying error state behavior for custom components, mostly. I have a few components for multi-selection that use both `required` and `minlength`, and these two would  result in the same error message for the end user.
Another use case is that I have a custom input component that has multiple invalid states, depending on user input. Invalid format, too short, too long, etc are all different error classes - and *some* of our screens don't really care about showing a different message per error class, but others do. So the different error classes are necessary, but the result is that there are lot of `hasError` calls or in some cases we implement an internal function just so these `hasError`s are simplified.

In other words, the motivation is simply convenience.

Thank you
## Environment

<pre><code>
Angular version: ^6.1.3
</code></pre>
